src/java/org/apache/cassandra/cql3/functions/VectorFcts.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.cql3.functions;
+
+import java.nio.ByteBuffer;
+import java.util.List;
+
+import org.apache.cassandra.cql3.CQL3Type;
+import org.apache.cassandra.db.marshal.AbstractType;
+import org.apache.cassandra.db.marshal.FloatType;
+import org.apache.cassandra.db.marshal.VectorType;
+import org.apache.cassandra.exceptions.InvalidRequestException;
+import org.apache.cassandra.transport.ProtocolVersion;
+import org.apache.lucene.index.VectorSimilarityFunction;
+
+public class VectorFcts
+{
+    public static void addFunctionsTo(NativeFunctions functions)
+    {
+        functions.add(similarity_function("similarity_cosine", VectorSimilarityFunction.COSINE, false));
+        functions.add(similarity_function("similarity_euclidean", VectorSimilarityFunction.EUCLIDEAN, true));
+        functions.add(similarity_function("similarity_dot_product", VectorSimilarityFunction.DOT_PRODUCT, true));
+    }
+
+    private static FunctionFactory similarity_function(String name,
```

Comment:
can we rename the function name to similarityFunction?

Reply:
I have renamed it to `createSimilarityFunctionFactory`.